### ISSUE TITLE
Add an option to DevTools to enable double-logging

### ISSUE THREAD

**Issue by gaearon** [facebook/react#19666]
We currently plan to suppress double-logging in development by overriding console in 17: https://github.com/facebook/react/pull/18547

However this can be confusing for some scenarios and double logging may be preferable in them.

Ideally we'd add a toggle to DevTools that would allow to re-enable it on this screen:

![Ef3bFlWWoAARjz9](https://user-images.githubusercontent.com/810438/90825360-b7372000-e330-11ea-9128-803d35d27e07.png)

**Comment by Jchinonso** [comment]
I could take this up!

**Comment by gaearon** [comment]
Sounds good.

**Comment by Jchinonso** [comment]
It seems there is a revamp of the UI, a checkbox 
- [ ] Break on warning 

After testing, i am yet to know exactly what it does. What do you think a suitable name will be e.g - Activate console warnings and errors,  -Enable double logging 
![Screenshot from 2020-08-25 19-26-05](https://user-images.githubusercontent.com/23220841/91213243-364d9f00-e709-11ea-90d5-e2c9dd840a92.png)

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
do you mind helping out with ways to test this?

**Comment by Jchinonso** [comment]
@gaearon i need more light on how to go about this. I have created the UI already https://github.com/facebook/react/pull/19710 but pretty confused on how to get the boolean value in packages/shared/ConsolePatchingDev.js.

**Comment by todortotev** [comment]
@Jchinonso Are you working on the PR. If you are busy do you mind if I take it up?

**Comment by Jchinonso** [comment]
@todortotev PR is up already https://github.com/facebook/react/pull/19710

**Comment by todortotev** [comment]
> @todortotev PR is up already #19710

Apologies! My bad! Good work!

**Comment by Jchinonso** [comment]
> > @todortotev PR is up already #19710
> 
> Apologies! My bad! Good work!

No, worries

**Comment by zhangenming** [comment]
so the pr when will  merge and release, thanks

**Comment by SaNsK11** [comment]
@gaearon I'd like to contribute. Since I am a beginner could you provide some insight as in how to go about it?

**Comment by bisubus** [comment]
@stale It' still relevant

**Comment by eps1lon** [comment]
Fixed by https://github.com/facebook/react/pull/22030
TL;DR:
- Disable console suppression by default in React.
- DevTools should grays out (but not hide) logs during second render pass by default.
- DevTools has a new Debugging preference that allows users to suppress second logs entirely (mimicking the current default behavior).
